File: data structures and algorithms/cpp/05_019.cpp
// Using STL queue
#include <queue>
std::queue<std::string> queue;
queue.push("first");         // Enqueue
queue.push("second");        
std::string first_item = queue.front(); // View front element
queue.pop();                 // Dequeue
bool is_empty = queue.empty();

// Using deque for more flexibility
#include <deque>
std::deque<std::string> dq;
dq.push_back("first");       // Enqueue at back
dq.push_back("second");
std::string first = dq.front(); // View front
dq.pop_front();              // Dequeue from front
bool empty = dq.empty();

File: data structures and algorithms/cs/05_019.cs
using System;
using System.Collections.Generic;
using System.Collections.Concurrent;

public class MainClass {
    public static void Main(string[] args) {
        // Main method code goes here
    }
    
    public void QueueExamples() {
        // Using Deque (preferred for performance)
        Deque<string> queue = new Deque<string>();
        // Enqueue items
        queue.AddLast("first");
        queue.AddLast("second");
        // Dequeue (returns "first")
        string firstItem = queue.RemoveFirst();

        // Using Queue for thread-safe operations
        ConcurrentQueue<string> threadSafeQueue = new ConcurrentQueue<string>();
        // Enqueue items
        threadSafeQueue.Enqueue("first");
        threadSafeQueue.Enqueue("second");
        // Dequeue
        threadSafeQueue.TryDequeue(out string firstThreadSafeItem);
        // Check if empty
        bool isEmpty = threadSafeQueue.IsEmpty;
    }
}
